.husky/pre-commit

```diff
@@ -18,8 +18,12 @@ npm run lint:fix
 
 npm run gen:schema
 
-# Generate table of contents for markdown files in root directory
-npm run update:toc
+# Check if any markdown files have been changed
+if git diff --cached --name-only | grep -E '\.md$'
+then
+  # Generate table of contents for markdown files in root directory
+  npm run update:toc
+fi
 
```

Comment:
don't add unrelated changes like these to the PR, because there's no context on what's being changed here and why